Sorts/CountingSort.js

```diff
@@ -8,31 +8,38 @@
  * Animated Visual: https://www.cs.usfca.edu/~galles/visualization/CountingSort.html
  */
 
-function countingSort (arr, min, max) {
-  let i
-  let z = 0
+const countingSort = (arr, min, max) => {
+  // Create an auxiliary resultant array
+  const res = []
+  // Create the freq array
   const count = []
+  let len = max - min + 1
 
-  for (i = min; i <= max; i++) {
+  for (let i = 0; i < len; i++) {
     count[i] = 0
   }
-
-  for (i = 0; i < arr.length; i++) {
-    count[arr[i]]++
+  // Populate the freq array
+  for (let i = 0; i < arr.length; i++) {
+    count[arr[i] - min] += 1
   }
-
-  for (i = min; i <= max; i++) {
-    while (count[i]-- > 0) {
-      arr[z++] = i
-    }
+  // Create a prefix sum array out of the freq array
```

Comment:
All of the later code can be simplified into a few lines:
```js
for (const i in count)
  res.concat((new Array(count[i])).fill(i + min))

return res

Reply:
In this change, I have a problem. 

- The change which you mentioned does not ensure the stability of count sort. Consider for instance, we have 1000 students. There marks are in the range 0 to 100. Now we have the sort the students based on marks. If two or more students have same marks then there roll number order should follow. But the change which you mentioned will not guarantee this. That is the main point of my PR, ensuring stability to count sort. 

- The second fix which I can think of is, it is always a good practice to return the original array where the sorting takes place. I know it won't makes difference, but generally if we are implementing an outplace algorithms, it is generally recommended to copy the contents of auxiliary array to the original array. That is the reason I didn't returned the `res` array, instead I copied the contents of `res` array to original input array `arr`.

Please do let me know If I was not clear.

Reference: [Stability of count sort](https://en.wikipedia.org/wiki/Counting_sort#Input_and_output_assumptions)

Reply:
Hi @raklaptudirm,
I updated the code with the changes mentioned by you. Please review the commit and merge the PR.
Thanks